src/algebra/commute.lean

```diff
@@ -0,0 +1,230 @@
+/-
+Copyright (c) 2019 Neil Strickland. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Neil Strickland
+
+Commuting pairs of elements in not-necessarily-commutative monoids,
+groups and (semi) rings.  Centralizers as submonoids, subgroups or
+subrings.
+-/
+
+import algebra.group_power
+  group_theory.submonoid group_theory.subgroup
+  ring_theory.subring
+
+section monoid
+
+variables {M : Type*} [monoid M]
+
+@[reducible] def commute (a b : M) : Prop := a * b = b * a
+
```

Comment:
If you start a namespace after this definition, then you don't have to prepend every name with `commute.`.

Reply:
I thought about this, but the suffixes are very generic things like `refl`, `add`, `eq` and so on, and I think that it is confusing to write them with no prefix.

Reply:
Nevertheless, it's pretty standard across mathlib to use the namespace, even for lemmas with names like `ext` and `refl`.